src/python/pcmdi/io.py

```diff
@@ -31,3 +38,249 @@ def __init__(self, root, var, obs_dic, reference="default", file_mask_template=N
         self.filename = ref["filename"]
         self.reference = obs_name
         self.variable = var
+
+
+class JSONs(object):
+
+    def __init__(self, files):
+        self.data = {}
+        self.json_version = "2.0"
+        if len(files) == 0:
+            raise Exception("You need to pass at least one file")
+        for fnm in files:
+            # print fnm
+            f = open(fnm)
+            tmp_dict = json.load(f)
+            json_version = tmp_dict.get("json_version", None)
+            if json_version is None:
+                # Json format not stored, trying to guess
+                R = tmp_dict["RESULTS"]
+                out = R[R.keys()[0]]  # get first available model
+                out = out["defaultReference"]  # get first set of obs
+                k = out.keys()
+                k.pop(k.index("source"))
+                out = out[k[0]]  # first realization
+                # Ok at this point we need to see if it is json std 1 or 2
+                # version 1 had NHEX in every region
+                # version 2 does not
+                if "bias_xy_djf_NHEX" in out[out.keys()[0]].keys():
+                    json_version = "1.0"
+                else:
+                    json_version = "2.0"
+            # print "\tv%s" % json_version
+            if json_version == "1.0":  # ok old way we need to convert to 2.0
+                for model in R.keys():  # loop through models
+                    # print "Reading in model:",model
+                    m = R[model]
+                    # print "FIRST M:",m.keys()
+                    for ref in m.keys():
+                        aref = m[ref]
+                        if not(isinstance(aref, dict) and "source" in aref):  # not an obs key
+                            continue
+                        # print "\treading in ref:",ref
+                        # print aref.keys()
+                        reals = aref.keys()
+                        src = reals.pop(reals.index("source"))
+                        for real in reals:
+                            # print "\t\treading in realization:",real
+                            # print real
+                            areal = aref[real]
+                            areal2 = {"source": src}
+                            for region in areal.keys():
+                                # print "\t\t\tREGION:",region
+                                reg = areal[region]
+                                if region == "global":
+                                    region2 = ""
+                                else:
+                                    region2 = region + "_"
+                                areal2[region2 + "global"] = {}
```

Comment:
@doutriaux1 should these be hard-coded? I think using the region list (which is growing) might be a better path forward see share/default_regions.py